### PR TITLE
Refactor the Load Generator and Support SteadyState Types

### PR DESCRIPTION
**Pull request by rthallisey**
Refactor the load generator code.

- Define workload types as an interface so it can be extended to accommodate each job type
- Add support for steady-state job
- Simplify the config (rely on outside scripts to handle ns creation, iterations, ect...)

Test below shows a steady-state workload with `churn: 2` `count: 5`.  This will create 5 VMIs and delete 2 VMIs frequently over a `5m` period.
```
name: kubevirt-steady-state-test
timeout: 5m
count: 5
churn: 2
minChurnSleep: 20s
type: "steady-state"
object:
  templateFile: vmi-ephemeral.yaml
  inputVars:
    containerPrefix: quay.io/kubevirt
    containerImg: cirros-container-disk-demo
    containerTag: ""
    namespace: default
```
Screen Shot 2022-02-22 at 9 31 36 AM

**Release note**:
```release-note
Support steady-state job types in the load-generator tool
```

Future work:
- The SteadyState `CreateWorkload` is really the same as Burst `CreateWorkload`.  Call Burst create from SteadyState.
- Refacore the Watcher package

### CODE REVIEW

**Comment by rthallisey** [comment]
cc @marceloamaral @davidvossel

File: tools/perfscale-load-generator/object/unstructured.go
Comment (marceloamaral) [inline tools/perfscale-load-generator/object/unstructured.go:30]: Probably a left over to be removed

**Review by marceloamaral — CHANGES_REQUESTED** [review]
This PR will affect two prow jobs. So before merging this, you also need another PR to fix those jobs.

[periodic-kubevirt-performance-cluster-100-density-test](https://github.com/kubevirt/project-infra/blob/163333c8ef4085a6a8805a05d610879b34c69d15/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L792)

[periodic-kubevirt-performance-cluster-scale-density-test](https://github.com/kubevirt/project-infra/blob/163333c8ef4085a6a8805a05d610879b34c69d15/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L792)

Both jobs execute the [automation/perfscale-test.sh](https://github.com/kubevirt/kubevirt/blob/main/automation/perfscale-test.sh), which run the perfscale-load-generator tool. You also need to update this script accordingly to the new logic.

/hold

File: tools/perfscale-load-generator/config/config.go
Comment (rthallisey) [reply tools/perfscale-load-generator/config/config.go:85]: I wanted to reduce config options.  I default to `20` which should be plenty for the time being.

**Comment by rthallisey** [comment]
created https://github.com/kubevirt/project-infra/pull/1946.  I think this change can go in first and 1946 can quickly follow.

File: tools/perfscale-load-generator/config/config.go
Comment (rthallisey) [reply tools/perfscale-load-generator/config/config.go:87]: Changed both to unlimited.  We can come back and change this as the tool matures.

**Comment by rthallisey** [comment]
rebased.  @marceloamaral need another lgtm from you

**Pull request merged**
